### ISSUE TITLE
Test publisher playlist page mockups

### ISSUE THREAD

**Issue by KubaMikolajczyk** [Joystream/atlas#2058]
To be decided if an idea for this test is viable - @dmtrjsg 

# What we want to test
1️⃣ We want to see which version of the playlist management page will be more useful to the users and which one will they find more approachable to use.
&
2️⃣We want to test creating a new playlist with our new playlist flow
&
3️⃣ We want to test whether a custom thumbnail feature is a needed & useful 

# How we want to test it
We can do it in two ways:
We will perform moderated interviews, observe users' behavior while interacting with prototypes and ask users which they prefer and why.

# How we will measure results
We will 
- measure the success rate of given tasks
 - asses any viable comments and suggestions.
 - spot any pain points and potential pitfalls 

# Possible questions
- Have you ever used a playlist feature on a video platform? Which? How often?
- Which layout do you find more usable for you? Why?
- Is there any information missing on those screens? What exactly?
- When creating a playlist what would you want its thumbnail to display?
- Could you rate 1-5 potential values that you see in uploading your own custom thumbnail. Why?

# Test details
Time: Monday 07 Feb 22

**Comment by dmtrjsg** [comment]
@KubaMikolajczyk  agreed to incorporate to the pending moderate test on Monday  the 7th.

**Comment by KubaMikolajczyk** [comment]
#### Test overview
# Playlists 07.02.22
- Overview: We decided to run moderated tests to catch all potential pitfalls and eliminate all bugs in playlist page design and creating playlist flow.
- Participants: 4️⃣

# Testers to be paid 👇
- 5FvpRF7KGWG2jiwC4jcxoeziieHsaKUDubrE2gLJwhNUDLye
- 5HVxLAskxQZiQMJwfMWDGQfK4beevP17zjYm8PAaA7GTZqdW
- 5CfjBwVfBxrGxNF6PWt9Fh5A6KicPMptbR3AQ9MUUFkzsMSm
- one tester didn't have a wallet / will provide in future

# Test summary & Recommendations
![image](https://user-images.githubusercontent.com/31144205/152988831-aca179a0-5d9c-43a2-9457-aea61ad1f944.png)
### In the case of publisher playlist page layout the "list" layout was voted as a preferred one by all 4 testers
> playlists statuses are much easier to discover in the list view

> The second version is more data-rich. I don’t know much about playlists in the first view, even though it seems like there’s the amount of info is almost the same.

> I like the second one as it’s easier to scan visually and would probably require fewer clicks from me”
- In list view, additional feedback would be nice upon hovering tile to describe the action

![image](https://user-images.githubusercontent.com/31144205/153017885-20e14a3a-84de-4a19-ad3a-9eea67ac4f11.png)
### If it comes to creating a new playlist
- All testers completed the flow successfully
- All testers described the flow as easy:
> “It’s all very simple. Not a rocket science for sure.”
- When the page is in its empty state we should reconsider having two same "add video" action btns as it was described as weird and confusing - maybe we can hide add videos btn and leave just the field
- Field for adding new videos in an empty state shouldn't be in dotted stroke as it indicates that you can drop something on it
- All testers described custom thumbnails for the playlist as a 100% useful and needed feature (high value) - therefore we should definitely explore this concept
- In add videos panel we could use an option to select all and a counter of how much was selected

# Additional ideas & found issues
- One user mentioned that he finds adding video to the playlist while uploading new videos useful & desired
- Concept of unlisted videos - the wording itself - is not clear and users can't identify the concept itself just based on the name of the status (need additional explanation). The concept though becomes clear once see the radio option card description where it says: Visible with link only. We might want to reconsider the naming of that feature - may be to private? or any other that would be more clear to the users
- One user said that he likes using playlists as a way to discover new and interesting content on the subject - maybe just a hint for us to focus on this aspect of playlist more in future experiments.
- Users seem to be interested in seeing statistics for their playlists - maybe possible future improvement
- One user when recalling his experience with youtube mentioned video tags as a useful feature that he liked


--------------------------------------------------------------------------------------------------------

# TLDR. 🛑 🛑 🛑 🛑 
Below you can find a long test description  you don't have to read it to get a bare summary of tests - just read the information above 👆

#Interview
#### Have you uploaded any content to a video platform in the past?
- 1️⃣ Tester: ✅  (Youtube brand channel)
- 2️⃣ Tester: ✅  (Youtube, Wrzuta.pl)
- 3️⃣ Tester: ✅  (Youtube gaming videos)
- 4️⃣ Tester: ✅ (Youtube, Facebook, Instagram)

#### Have you uploaded any videos to Joystream studio? 
- 1️⃣ Tester: ✅  
- 2️⃣ Tester: ❌
- 3️⃣ Tester: ❌
- 4️⃣ Tester: ❌

#### Have you created any playlists in the past on any platform?
- 1️⃣ Tester: ✅  (Youtube brand channel)
- 2️⃣ Tester: ❌ 
- 3️⃣ Tester: ✅  (Group his gaming videos)
- 4️⃣ Tester: ❌

# 🟡  **My playlists page prototype test**
![image](https://user-images.githubusercontent.com/31144205/152988831-aca179a0-5d9c-43a2-9457-aea61ad1f944.png)
During this part of the test, we asked testers to interact with two different versions of the same page and describe for us all actions that they can do with them.

# 1️⃣ Tester
### First layout "Grid"
- Recognised that he can play playlist by pressing on a thumbnail
- Knows where to edit the playlist
- Did not notice the context menu 
> It wasn’t clear, because I had to hover particularly the title. 

### Second layout "List"
- Instantly noticed all available options.
> In the previous prototype the problem was I wasn’t hovering over the title.
- Recognises why he didn't discover the context menu in the previous prototype:
> As a flow, I always want to hover over the picture. The picture is always more interactive than text
- When hovering single row in List layout:
> I like that I can hover anywhere
- When hovering over the row he said:
> Over here it’s not clear if I can play a playlist just by clicking over the thumbnail. In the tiles view it’s clear because there’s a play icon in the middle. But in the list view, while it’s cool I have all the options appearing, I would intuitively click on the picture expecting the playlist to be played.

### How do you think which version of those screens would you use more? why?
> I think I would use the tiles one, as a creator.
>For the files system, on my mac, I would probably use the list view.
Tiles feel a little bit cozier. Somehow I like it more.
Again stresses out about the dotted area. He says he likes it a lot.

>On the other hand, I like that I have everything in one place in the list view.

### Is there any information missing on those screens that you would like to see there?
>If you have a massive amount of playlists, it’s obviously easier to manage them using the list view.
Maybe I would appreciate a bigger “Create playlists” CTA, just like on the tiles view.

### Is there anything here that you wouldn’t use at all?
- He struggles to answer this one.
- Coming back to the list view, he says it’s much more informative. 
⚠️ **USER CHANGES HIS MIND**
- Now he says he likes the list more.
- He likes that he can easily sort things and that he can see everything at a glance (cause it’s a table).
>If you think about how we’re reading, left to right and in lines, the structure of the data is easier on the eye. Birds-eye view.
>Whereas in the tiles view I need to jump around with my eyes in order to process the information

### 🟢 **Summary**
The user started by stating that he likes tiles more, but really quickly changes his mind and leans toward a list solution saying that it allows him to scan information more efficiently.



# 2️⃣ Tester
### First layout "Grid"
- He imagines he could drag and drop playlist tiles to reorder them.
- He imagines he can just click on a playlist thumbnail to play it.
- He would appreciate having a quick preview of the playlist’s content without having to open it.
- He has managed to discover all basic functionalities.

### Second layout "List"
- Successfully spotted the differences between A and B.
- He claims this is more transparent.
> To me, as a viewer, tiles would be probably easier, but as a publisher, the list view is probably more useful
- Describes this layout as:
> An admin panel for a playlist manager
-Seems to understand each column’s meaning.

### How do you think which version of those screens would you use more? why?
- The list view, without a doubt
>Eg. playlists statuses are much easier to discover in the list view

### Is there any information missing on those screens that you would like to see there?
>On the tiles view, the date of publication is not super easy to discover.

>I would appreciate the number of views for ... playlists/videos(in total?).

> As someone who is interested in earning money from that, I would probably appreciate having stats for playlists.

### Is there anything here that you wouldn’t use at all?
>I don’t know, I would probably need to see how does it look like with 100 playlists to really tell if the tabs above are something I would really use

### 🟢 **Summary**
This user is strongly convinced that a list view would allow him to scan content more easily.


# 3️⃣ Tester
### First layout "Grid"
-Doesn’t know what “Unlisted” means, his best guess is that it might have something to do with privacy settings, but also thinks this might be connected with NFTs
- He understands that by pressing on a tile thumbnail he can start watching it.
- He notices the edit button and understands what it does.
- He notices the context menu.
- He struggles to understand what the time duration is about. His best guess is that it might be a total watching time, but he thinks it could be also something else.
- He understands what a Draft is.
- He notices sorting options and where to look for a playlist creation button (or area).
- When asked about how to delete a playlist, he answers that he would go for a dropdown list (which is correct).

### Second layout "List"
>Admin panel-like view
- He notices that these are the same playlists but notices that it feels like there is more info present at a glance.
- Notices the visibility column right away. Still struggles with understanding the “Unlisted” setting label.
- He notices playlists options right away and names all of them correctly.
- He wouldn’t expect anything to happen upon clicking on a playlist thumbnail in this view. But when asked for his best guess, he said he suspects this would allow him to play the playlist.

### How do you think which version of those screens would you use more? why?
>The second version is more data-rich

> I don’t know much about playlists in the first view, even though it seems like there’s the amount of info is almost the same.

>I can get more out of the list view”, he says. “It’s more clear.

### Is there any information missing on those screens that you would like to see there?
- He mentioned the number of views for a playlist, without being super certain about it.
- But when asked about clarification on what exactly this would mean, he wasn’t able to provide a clear answer.

### Is there anything here that you wouldn’t use at all?
- He noticed that the number of videos is duplicated information, thus he believes one of them is unnecessary.

### 🟢 **Summary**
This tester described the list version as more data-rich even though, as he noticed himself, the amount of data is similar on both screens. He said he can get more from the second layout.


# 4️⃣ Tester
### First layout "Grid"
> I can see there’s something happening when I hover over them

> It’s weird that the context menu button appears only upon hovering the title
- He noticed the Edit playlist button right away. And the context menu, too.

>Not sure how can I change the status nor what does it exactly mean. Ok, <<Draft>> is rather clear to me, but <<Unlisted>>? Maybe it means that it’s hidden from the public, but on the other hand, can be similar to the Draft

### Second layout "List"
>Ok, this is much clear now — visibility. Because I know what it refers to...

>Unlisted... Maybe the label is wrong? Maybe it should be a different label?
- Notices actions right away.
>Ok, I see here much more information than on the previous view.
- Notices the total time info right away and claims he didn’t see that in the previous view (he did).
>Pretty straightforward

>When I click on the thumbnail I’m assuming it will open details of the playlists with all the videos, the actions you can do, and the additional information

### How do you think which version of those screens would you use more? why?
>Second one.

> the First one is maybe more visually appealing, as it’s cleaner, but eg. the total time info is something I totally missed [on the tiles view].

>On the tiles view: “I had a feeling I can drag and drop something over the big dotted “Create new playlists” button.”

>I like the second one as it’s easier to scan visually and would probably require fewer clicks from me”.

### Is there any information missing on those screens that you would like to see there?

- Nothing that I can think of.

### Is there anything here that you wouldn’t use at all?

- Thinks for a long while
> Maybe bulk options for deleting, I’d say?
> (About the icons on the list view) “Labels are a bit misleading maybe?

### 🟢 **Summary**
The user claims that the second screen is better for him.

# 🟡  **Creating playlist prototype**
![image](https://user-images.githubusercontent.com/31144205/153017885-20e14a3a-84de-4a19-ad3a-9eea67ac4f11.png)
During this part of the test, we asked testers to create a new playlist from scratch. We wanted to check if every feature is readable and easy to use.

# 1️⃣ Tester
### While going through the prototype
- Wonders what’s the difference between the two “Add videos” buttons/areas. 
- Adds title and description without any problems.
- Instantly notices the “Publish playlist” button, presses it instantly.

### How was it for you to create a playlist that way?
> “It was okay”.

>” I didn’t understand that’s the difference between the two <<Add videos>> buttons. They’re in both places, but they mean the same thing.”

### Please tell me would you ever consider uploading your own custom thumbnail for a playlist. Why?

> Definitely! 100%

> You can have different videos on different topics, if you have 10 videos in a playlist, then it’s okay it’s natural you could use one of the thumbnails you already had, but maybe sometimes you would want to upload a custom thumbnail, to make it more appealing for people to click it.

> Maybe sometimes the videos you had are so different, that none of the thumbnails is good enough to capture the true playlist’s content.

> I wouldn’t say I missed this option in YouTube, but still to me it’s so natural you can have a custom thumbnail for a playlist

### Did you have any problems at any point in the flow or do you find something unclear?

- Again, mentions the “Add videos” buttons and how confusing they were (or the fact, that they are both present at the same time, very close to each other, doing the same thing).



# 2️⃣  Tester
### While going through the prototype
> I would start by clicking on the blue button. The other one is harder to see
- It’s easy for him to figure out how to add new videos.
- Sets up a thumbnail easily.
- Then he reorders videos expecting for this to affect the thumbnail, but since he already set it up in the dropdown before, he assumes it’s simply not possible
- He moves buttery smooth through the rest of the process. No issues. Successfully published the playlist.
>Instinctively, I would assume that the first video is the one that is affecting the playlist’s thumbnail

### How was it for you to create a playlist that way?
> Just as I expected. Nothing came as a surprise to me. It was simple

### Please tell me would you ever consider uploading your own custom thumbnail for a playlist. Why?
- He believes it would be useful for channel owners to be able to upload their own thumbnails.
- As an example, he mentioned that as an artist he might want to upload a CD cover as a playlist thumbnail.

### Did you have any problems at any point in the flow or do you find something unclear?
- Claims it wasn’t super clear from the beginning how to set up a thumbnail. He did this without a hustle anyway, so yeah, let’s be careful.

# 3️⃣Tester
### While going through the prototype
- He reaches for the blue button to create a playlist as a first thing but ends up clicking on the creation area.
> I would love to be able to select all videos at once
- When asked to change a thumbnail, he first reached for a thumbnail area, and once realized it was not working (in the prototype), he reordered videos and moved “Ghost Signals” to the top in order to change the playlist’s thumbnail.
- Was struggling a bit to find a playlists title area.
- At this point, he discovers what does “Unlisted” means.
- He wouldn’t look for thumbnail changing options in dropdown menus.
- He said he would be surprised to see a thumbnail not changing upon reordering videos, even though he previously changed it from the dropdown menu.

### How was it for you to create a playlist that way?
> Overall, okay. I missed a select all button when adding videos If there were 20 videos, it would take me a while to select all of them, right?

### Please tell me would you ever consider uploading your own custom thumbnail for a playlist. Why?
> It depends. In my case, I would probably upload a game logo to indicate the playlist’s contents being about videos on a particular video game. I could see playlist thumbnails visually corresponding to video thumbnails feel & look.

### Did you have any problems at any point in the flow or do you find something unclear?
> No. Just the thumbnail setting experience. And I would also expect to change the thumbnail in the UI somewhere around the thumbnail area


# 4️⃣Tester
### While going through the prototype
- He’s trying to set up a thumbnail as the first thing.
- Finally discovers the context menu and sets the thumbnails from there.
- Adds the rest of the info effortlessly.
- Now he’s having his “aha moment” regarding what does “Unlisted” means.

### How was it for you to create a playlist that way?
> The first thing I did after entering the adding playlists view was look all the way to the left-hand side [where you can set up a playlist’s thumbnail], whereas I should be probably guided towards the right side, as this is where you can click to add a video

> Other than this it was simple and straightforward

### Please tell me would you ever consider uploading your own custom thumbnail for a playlist. Why?
- Yes,
- Maybe none of the videos thumbnails would be relevant to the content of the entire playlists?


### Did you have any problems at any point in the flow or do you find something unclear?
none

**Comment by dmtrjsg** [comment]
Great job @KubaMikolajczyk 🙌 

Comments:

1. Statistics -> would be good to add the total counter of videos watched onto the playlist level. This would help us select some playlists to popular section on home page and discover.
2. Re custom Thumbnail - this is pretty key for content discoverability (optimisation of it). If we do it, let's scope out the best possible way to achieve this, and perhaps add this as a follow up feature, based on dev efforts.

For both comments @kdembler's input is key to make a decision :) 

other: let's pay all users in a single go, so pls ping me when the list is complete ;)

**Comment by dmtrjsg** [comment]
Was decided on the meeting 9th Feb counter for playlist level will be skipped for now. 
Thumbnail will be taken from first video by default, and can be customised - replaced with a custom image.
